pipelines/common/src/main/java/org/openmrs/analytics/DwhFiles.java

```diff
@@ -154,39 +178,96 @@ public Set<String> findResourceTypes() throws IOException {
     return typeSet;
   }
 
+  /**
+   * This method copies all the files under the directory (getRoot() + resourceType) into the
+   * destination DwhFiles under the similar directory.
+   *
+   * @param resourceType
+   * @param destDwh
+   * @throws IOException
+   */
   public void copyResourcesToDwh(String resourceType, DwhFiles destDwh) throws IOException {
-    Path path = fileSystem.getPath(getRoot(), resourceType);
-    Path destPath = destDwh.getResourcePath(resourceType);
-    log.info("Copying {} to {}", path, destPath);
-    FileUtils.copyDirectory(path.toFile(), destPath.toFile());
+    ResourceId sourceFiles =
+        FileSystems.matchNewResource(getRoot(), true)
+            .resolve(resourceType, StandardResolveOptions.RESOLVE_DIRECTORY)
+            .resolve("*", StandardResolveOptions.RESOLVE_FILE);
+
+    List<MatchResult> sourceMatchResultList = FileSystems.matchResources(List.of(sourceFiles));
+    List<ResourceId> sourceResourceIdList = new ArrayList<>();
+    for (MatchResult matchResult : sourceMatchResultList) {
+      if (matchResult.status() == Status.OK) {
+        List<ResourceId> resourceIds =
+            matchResult.metadata().stream()
```

Comment:
Question: In what cases `matchResult.metadata()` has a cardinality higher than 1?

Reply:
I assume that the number of parquet files for any resource type could be like ./Observation/*.parquet which can be more than 1. For example, during the initial batch run my test data had these many records Observation=15700, Encounter=4500, Patient=85. And the default jdbcFetchSize was set to 1000. So when the batch run completed it created 16 Observation parquet files, 5 Encounter files and 1 Patient file. So I feel for Observation and Encounter the cardinality can be > 1.

Reply:
Yes, `sourceMatchResultList` will have only one element since in the initial matching only one input element was used.
List<MatchResult> sourceMatchResultList = FileSystems.matchResources(List.of(sourceFiles));

Comment added.